src/EfficientDynamoDb/Internal/Core/Utilities/ThreadSafeRandom.cs

```diff
@@ -10,6 +10,6 @@ internal sealed class ThreadSafeRandom : IRandom
 
         public static readonly ThreadSafeRandom Instance = new ThreadSafeRandom();
 
-        public int Next(int maxValue) => Random.Value.Next(maxValue);
+        public int Next(int maxValue) => Random.Value!.Next(maxValue);
```

Comment:
I'd suggest to remove this type (and IRandom) in the separate PR, and use `Random.Shared` instead.

Reply:
I agree that `Random.Shared` should be used. But I don't think that we can remove the interface because it will make impossible to test retry strategies.

Reply:
Well, `Random` type has virtual methods, so it's still possible to mock it without an interface.

Reply:
@Dreamescaper Oh, I didn't realize that methods in `Random` are virtual. In this case you're right and we can remove the interface as well.